### PR TITLE
Add users/lookup endpoint and move Facebook user creation to UsersController

### PR DESCRIPTION
**Pull request by joshsmith**
This PR attempts to change the authentication flow required by https://github.com/Bernie-2016/ground-game-ios/pull/37

We added a `users/lookup` endpoint to find a user by their Facebook ID.

We also removed creating a user from Facebook from `oauth/tokens`, and created a split method in `users#create`. This separates the concerns: finding a user and providing a token is handled by the `TokensController`. Creating a user is handled by the `UsersController`.

The API flow if the user is signing up from Facebook is:
- Client requests whether user exists from `users/lookup`
- If so, client requests authorization with Facebook token
- If not, client `POST`s to `users#create`, and on success, requests authorization with Facebook token

This still needs to be cleaned up with at least the following:
- [x] Add tests and code for not finding a user with their `facebook_id` in `TokensController`
  - Added the tests and the behavior, but I'm not sure what the error should say. 
  - `ErrorSerializer` now supports `Koala::Facebook::AuthenticationError`
- [x] Add tests for creating a user from facebook with a separate image (it should not use their facebook photo)
- [x] Add other edge case tests for new facebook user creation method
- [x] Possibly remove unneeded `UpdateUserAttributesFromFacebook`

### CODE REVIEW

File: spec/requests/api/users_spec.rb [hunk 1]
```diff
@@ -11,103 +11,381 @@
...
+          expect(user.email).to eq email
+          expect(user.first_name).to eq first_name
+          expect(user.last_name).to eq last_name
+          expect(user.state_code).to eq state_code
+          expect(user.lat).to eq lat
+          expect(user.lng).to eq lng
+          expect(user.facebook_id).to eq @facebook_user["id"]
+          expect(user.facebook_access_token).to eq @facebook_access_token
+
+          expect(InitializeNewFacebookUserWorker.jobs.size).to eq 1
```
Comment (joshsmith) [inline spec/requests/api/users_spec.rb:158]: What do we expect the `AddFacebookProfilePicture.jobs.size` to be?

**Comment by joshsmith** [comment]
:+1:
